src/Components/Form/Form.js

```diff
@@ -0,0 +1,327 @@
+    this.onReset = this.onReset.bind(this);
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
+  }
+
+  onChange = (event) => {
+    const name = event.target.name;
+    const value = event.target.value;
+    switch (name) {
+      case "username":
+      case "surname":
+        if (value && value.trim()[0]?.toUpperCase() !== value.trim()[0]) {
+          this.setState((state) => ({
+            formError: {
+              ...state.formError,
+              [name]:
+                name === "username"
+                  ? "Имя должно начинаться с заглавной буквы"
+                  : "Фамилия должна начинаться с заглавной буквы",
+            },
+          }));
+        } else {
+          this.setState((state) => ({
+            formError: {
+              ...state.formError,
+              [name]: "",
+            },
+          }));
+        }
+        this.setState((state) => ({
+          form: {
+            ...state.form,
+            [name]: value.trim(),
+          },
+        }));
+        break;
+      case "date":
+        if (value.length > 0) {
+          this.setState((state) => ({
+            formError: {
+              ...state.formError,
+              date: "",
+            },
+          }));
+        }
+        this.setState((state) => ({
+          form: {
+            ...state.form,
+            date: value,
+          },
+        }));
+        break;
+      case "phone":
+        if (/[a-zа-яё]/i.test(value[0])) {
+          return this.setState((state) => ({
+            form: { ...state.form, phone: "" },
+            formError: {
+              ...state.formError,
+              phone: "Номер телефона должен состоять только из цифр",
+            },
+          }));
+        }
+        const phone = value
+          .replace(/[^\d]/g, "")
+          .match(/(\d)(\d{0,4})(\d{0,2})(\d{0,2})/);
+        let formatted;
+        if (Array.isArray(phone)) {
+          formatted = phone
+            .slice(1, 5)
+            .filter((item) => !!item)
+            .join("-");
+        }
+        this.setState((state) => ({
+          form: { ...state.form, phone: formatted },
+          formError: { ...state.formError, phone: "" },
+        }));
+        break;
+      case "site":
+        const http = "https://";
+        if (http.length >= value.length && !http.startsWith(value)) {
+          return this.setState((state) => ({
+            formError: {
+              ...state.formError,
+              site: "Адрес сайта должен начинаться с https://",
+            },
+          }));
+        }
+        this.setState((state) => ({
+          form: {
+            ...state.form,
+            site: value.trim(),
+          },
+          formError: {
+            ...state.formError,
+            site: "",
+          },
+        }));
+        break;
+      case "about":
+      case "stack":
+      case "description":
+        if (value && value.length >= 600) {
```

Comment:
You can extract 'magic number' 600 to constant.

Reply:
done